broker/query_plan_agg.go

```diff
@@ -451,7 +457,7 @@ func buildSubPlan(agg common.AggType, qc QueryContext, assignments map[topology.
 	root := NewMergeNode(agg)
 	for host, shardIDs := range assignments {
 		// make deep copy
-		currQ := *qc.AQLQuery
+		currQ := qc.GetRewrittenQuery()
```

Comment:
can this line move to outside of the loop?

Reply:
not sure. its address is passed into the currQc, so I don't want to mess with that.

Reply:
likely make a copy of currQ instead of recalculation